src/ethereumWallet.ts

```diff
@@ -6,9 +6,11 @@ import { HDNode } from "ethers/utils/hdnode";
 export class EthereumWallet {
   private readonly wallet: Wallet;
 
-  public constructor(key: SigningKey | HDNode | Arrayish, jsonRpcUrl: string) {
+  public constructor(jsonRpcUrl: string, key?: SigningKey | HDNode | Arrayish) {
     const provider = new ethers.providers.JsonRpcProvider(jsonRpcUrl);
-    this.wallet = new ethers.Wallet(key).connect(provider);
+    const wallet = !key ? ethers.Wallet.createRandom() : new ethers.Wallet(key);
```

Comment:
```suggestion
    const wallet = key ? ethers.Wallet.createRandom() : new ethers.Wallet(key);
```
It is a strange use of the `!` to me here, can you explain?

Reply:
If the key is undefined because it wasn't passed as an argument, create a random wallet. Otherwise, create a wallet with the key passed as an argument.

Reply:
I see, I think the usage of `!` with a ternary operator just makes it less readable.
```suggestion
    const wallet = key ? new ethers.Wallet(key) : ethers.Wallet.createRandom();
```

Reply:
Fair point!